File: WinGo/src/main/java/Entites/Suggestion.java
package Entites;

import java.sql.Date;

public class Suggestion {
    private int id_suggestion;
    private int id_user;
    private String sujet;
    private String description;
    private String categorie;
    private Date date_suggestion;
    private String statut;
    private Integer id_reclamation;
    private String reponse_admin;
    private Date date_reponse;

    public Suggestion() {
    }

    public Suggestion(int id_user, String sujet, String description, String categorie, Integer id_reclamation) {
        this.id_user = id_user;
        this.sujet = sujet;
        this.description = description;
        this.categorie = categorie;
        this.id_reclamation = id_reclamation;
        this.statut = "Attente";
        this.date_suggestion = new Date(System.currentTimeMillis());
    }

    public int getId_suggestion() {
        return id_suggestion;
    }

    public void setId_suggestion(int id_suggestion) {
        this.id_suggestion = id_suggestion;
    }

    public int getId_user() {
        return id_user;
    }

    public void setId_user(int id_user) {
        this.id_user = id_user;
    }

    public String getSujet() {
        return sujet;
    }

    public void setSujet(String sujet) {
        this.sujet = sujet;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCategorie() {
        return categorie;
    }

    public void setCategorie(String categorie) {
        this.categorie = categorie;
    }

    public Date getDate_suggestion() {
        return date_suggestion;
    }

    public void setDate_suggestion(Date date_suggestion) {
        this.date_suggestion = date_suggestion;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    public Integer getId_reclamation() {
        return id_reclamation;
    }

    public void setId_reclamation(Integer id_reclamation) {
        this.id_reclamation = id_reclamation;
    }

    public String getReponse_admin() {
        return reponse_admin;
    }

    public void setReponse_admin(String reponse_admin) {
        this.reponse_admin = reponse_admin;
    }

    public Date getDate_reponse() {
        return date_reponse;
    }

    public void setDate_reponse(Date date_reponse) {
        this.date_reponse = date_reponse;
    }

    @Override
    public String toString() {
        return "Suggestion{" +
                "id=" + id_suggestion +
                ", sujet='" + sujet + '\'' +
                ", categorie='" + categorie + '\'' +
                '}';
    }
}

File: src/Entity/Suggestion.php
<?php

namespace App\Entity;

use App\Repository\SuggestionRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: SuggestionRepository::class)]
#[ORM\Table(name: 'suggestion')]
class Suggestion
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_suggestion', type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(name: 'id_user', type: 'integer')]
    private int $idUser = 0;

    #[ORM\Column(type: 'string', length: 100, nullable: true)]
    private ?string $sujet = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column(type: 'string', length: 50, nullable: true)]
    private ?string $categorie = null;

    #[ORM\Column(name: 'date_suggestion', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $dateSuggestion = null;

    #[ORM\Column(type: 'string', length: 30, options: ['default' => 'Recue'])]
    private string $statut = 'Recue';

    #[ORM\Column(name: 'reponse_admin', type: 'text', nullable: true)]
    private ?string $reponseAdmin = null;

    #[ORM\Column(name: 'date_reponse', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $dateReponse = null;

    #[ORM\Column(name: 'id_reclamation', type: 'integer', nullable: true)]
    private ?int $idReclamation = null;

    public function __construct() { $this->dateSuggestion = new \DateTime(); }

    public function getId(): ?int { return $this->id; }
    public function getIdUser(): int { return $this->idUser; }
    public function setIdUser(int $idUser): static { $this->idUser = $idUser; return $this; }
    public function getSujet(): ?string { return $this->sujet; }
    public function setSujet(?string $sujet): static { $this->sujet = $sujet; return $this; }
    public function getDescription(): ?string { return $this->description; }
    public function setDescription(?string $description): static { $this->description = $description; return $this; }
    public function getCategorie(): ?string { return $this->categorie; }
    public function setCategorie(?string $categorie): static { $this->categorie = $categorie; return $this; }
    public function getDateSuggestion(): ?\DateTimeInterface { return $this->dateSuggestion; }
    public function setDateSuggestion(?\DateTimeInterface $dateSuggestion): static { $this->dateSuggestion = $dateSuggestion; return $this; }
    public function getStatut(): string { return $this->statut; }
    public function setStatut(string $statut): static { $this->statut = $statut; return $this; }
    public function getReponseAdmin(): ?string { return $this->reponseAdmin; }
    public function setReponseAdmin(?string $reponseAdmin): static { $this->reponseAdmin = $reponseAdmin; return $this; }
    public function getDateReponse(): ?\DateTimeInterface { return $this->dateReponse; }
    public function setDateReponse(?\DateTimeInterface $dateReponse): static { $this->dateReponse = $dateReponse; return $this; }
    public function getIdReclamation(): ?int { return $this->idReclamation; }
    public function setIdReclamation(?int $idReclamation): static { $this->idReclamation = $idReclamation; return $this; }
}
